src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -259,5 +261,68 @@ private static CompletionItemRules MakeRule(bool importDirective, bool preselect
         }
 
         protected override CompletionItemSelectionBehavior PreselectedItemSelectionBehavior => CompletionItemSelectionBehavior.HardSelection;
+
+        protected override CompletionItem CreateItem(
+            CompletionContext completionContext,
+            string displayText,
+            string displayTextSuffix,
+            string insertionText,
+            List<ISymbol> symbols,
+            SyntaxContext context,
+            bool preselect,
+            SupportedPlatformData supportedPlatformData)
+        {
+            var rules = CreateCompletionItemRules(completionContext, symbols, context, preselect);
+            var symbol = symbols[0];
+            var item = SymbolCompletionItem.CreateWithNameAndKind(
+                displayText: displayText,
+                displayTextSuffix: displayTextSuffix,
+                symbols: symbols,
+                // Always preselect
+                rules: rules,
+                contextPosition: context.Position,
+                insertionText: insertionText,
+                filterText: GetFilterText(symbol, displayText, context),
+                supportedPlatforms: supportedPlatformData);
+
+            if (symbol.IsKind(SymbolKind.Method)
```

Comment:
I feel it might be easier to follow if this entire `if` statement is rewritten by a series of old fashioned if-else instead of switch expressions.